client/network/src/protocol/event.rs

```diff
@@ -67,7 +67,16 @@ pub enum Event {
 		/// Node we opened the substream with.
 		remote: PeerId,
 		/// The concerned protocol. Each protocol uses a different substream.
+		/// This is always equal to the value of
+		/// [`crate::config::NonDefaultSetConfig::notifications_protocol`] of one of the
+		/// configured sets.
 		protocol: Cow<'static, str>,
+		/// If the negotiation didn't use the main name of the protocol (the one in
+		/// `notifications_protocol`), then this field contains which name has actually been
+		/// used.
+		/// Always contains a value equal to the value in
+		/// [`crate::config::NonDefaultSetConfig::fallback_names`].
+		negotiated_fallback: Option<Cow<'static, str>>,
```

Comment:
Why don't you use `protocol: Either protocol fallback`?

Reply:
`protocol` is used as an identifier on the API level, which is why the API seems a bit weird maybe. It is always equal to the main protocol name passed by the user, so that the rest of the code doesn't have to compare it to multiple different values.